gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/structure/io/binary/types/EdgeSerializer.java

```diff
@@ -42,12 +44,15 @@ public EdgeSerializer() {
     @Override
     protected Edge readValue(final Buffer buffer, final GraphBinaryReader context) throws IOException {
         final Object id = context.read(buffer);
-        final String label = context.readValue(buffer, String.class, false);
+        // reading single string value for now according to GraphBinaryV4
```

Comment:
Nit: this comment makes it seem like GraphBinaryV4 forces labels to be a "single string", but that isn't the case.

Reply:
Updated